### PR TITLE
[receiver-mock] Add mocked endpoints for terraform fields API

### PR DESCRIPTION
**Pull request by pmalek-sumo**

### CODE REVIEW

File: src/rust/receiver-mock/src/router.rs [hunk 1]
```diff
@@ -167,6 +167,82 @@ pub async fn handler_terraform_fields_quota() -> impl Responder {
     })
 }
 
+pub async fn handler_terraform_fields() -> impl Responder {
+    #![allow(non_snake_case)]
```
Comment (sumo-drosiek) [inline src/rust/receiver-mock/src/router.rs:171]: why?

Comment (sumo-drosiek) [reply src/rust/receiver-mock/src/router.rs:171]: Do want to achieve this?
https://serde.rs/attr-rename.html

**Review by sumo-drosiek — COMMENTED** [review]
Generally LGTM, but using `non_snake_case` is not good approach for this case

File: src/rust/receiver-mock/src/router.rs (hunk 1 above)
Comment (pmalek-sumo) [reply src/rust/receiver-mock/src/router.rs:171]: image

File: src/rust/receiver-mock/src/router.rs [hunk 2]
```diff
@@ -167,6 +173,164 @@ pub async fn handler_terraform_fields_quota() -> impl Responder {
...
+    terraform_state: web::Data<TerraformState>,
+) -> impl Responder {
+    #[derive(Serialize)]
+    #[serde(rename_all = "camelCase")]
+    struct TerraformFieldResponse {
+        field_name: String,
+        field_id: String,
+        data_type: String,
+        state: String,
+    }
```
Comment (sumo-drosiek) [inline src/rust/receiver-mock/src/router.rs:222]: Duplication of https://github.com/SumoLogic/sumologic-kubernetes-tools/pull/75/files#diff-f45289341f4631ac823eb8b9fdc542eb2200f8d3c4734c554442f19cb3c1e539R179-R186

Comment (sumo-drosiek) [reply src/rust/receiver-mock/src/router.rs:222]: Please move it in common place

Comment (pmalek-sumo) [reply src/rust/receiver-mock/src/router.rs]: done

File: src/rust/receiver-mock/src/router.rs [hunk 3]
```diff
@@ -167,6 +173,163 @@ pub async fn handler_terraform_fields_quota() -> impl Responder {
...
+    terraform_state: web::Data<TerraformState>,
+) -> impl Responder {
+    #[derive(Serialize)]
+    #[serde(rename_all = "camelCase")]
+    struct TerraformFieldCreateResponse {
+        field_name: String,
+        field_id: String,
+        data_type: String,
+        state: String,
+    }
```
Comment (sumo-drosiek) [inline src/rust/receiver-mock/src/router.rs:281]: Please re-use `TerraformFieldObject`

Comment (pmalek-sumo) [reply src/rust/receiver-mock/src/router.rs]: done

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**